### ISSUE TITLE
NSX - Upgrade ncp in KLAB2 and EMERALD to v4.2.1

### ISSUE THREAD

**Issue by wmhutchison** [bcgov/developer-experience#5241]
**Describe the issue**
A version of ncp which supports upgrading to Openshift 4.16 is now available. Investigate and if no technical blockers exist, pursue upgrading ncp to this version in KLAB2 and EMERALD.

**Additional context**
[Link to ncp release notes.](https://docs.vmware.com/en/VMware-NSX-Container-Plugin/4.2.1/rn/vmware-nsx-container-plugin-421-release-notes/index.html)

**How does this benefit the users of our platform?**
Newer software supporting allowing NSX clusters to upgrade via EUS to OCP 4.16.

**Definition of done**
- [x] Confirm technical dependencies for current OCP and NSX versions.
- [x] Download new ncp/operator and upload to Artifactory.
- [ ] Upgrade ncp in KLAB2 and allow bake-in time.
- [ ] Schedule and implement an upgrade of ncp in EMERALD.

**Comment by wmhutchison** [comment]
FYI, latest ako for AVI currently supports up to OCP 4.15: https://docs.vmware.com/en/VMware-Avi-Load-Balancer/1.12/Avi-Kubernetes-Operator-Release-Notes/GUID-D2E95B86-A46A-48AD-9A1D-01115BD58D41.html

**Comment by wmhutchison** [comment]
Can't check release notes page, since it's currently broke, both what's linked in this ticket and what Google finds, will re-check later.

Just finished uploading new images to Artifactory. Looking through the big zip file to see if the vendor has updated manifests we should be using for things like Deployment versus DeploymentConfigs.

**Comment by wmhutchison** [comment]
NSX is going to be upgraded from v3.2.3 to v3.2.4.1. The current ncp will not suffice for that and the proposed original ncp version is too new. Thus an intermediate version of ncp will be required and also upgraded/applied at the same time as when this NSX upgrade is to be applied.

Thus update this ticket to reflect we're pushing for an upgrade to ncp v4.1.2.2 instead which will weather us through the NSX planned upgrade. Once that's done (NSX upgrade and ncp to v4.1.2.2), we'll be able to freely plan the OCP 4.16 upgrade along with ncp v4.2.1 (or newer version if applicable) when we're ready for that.

**Comment by wmhutchison** [comment]
I still want this ticket, so going to Backlog it, create a new ticket for aligning with NSX upgrade work, and make this one dependent on the new ticket.

**Comment by wmhutchison** [comment]
Ticket is now Backlogged again. Creating a new ticket with more immediate priority involving ncp upgrade to v4.1.2.2 instead, which will need to align with VMWare's NSX upgrade work. Only then can the other ncp upgrade ticket be tendered for execution.

**Comment by wmhutchison** [comment]
For clarity I've added a dependency blocker between this ticket and another for upgrading ncp to v4.1.2.2. Once that initial upgrade is done and tested, we're unblocked for moving forward with this ticket which will unblock us partly for OCP 4.16 upgrade in KLAB2/EMERALD.

**Comment by wmhutchison** [comment]
Closing this ticket out in favor of the official ticket(s) which are part of the bundle when doing official OCP upgrades. There's a newer version than the one listed in this ticket as well.
